refactoring and fix

ImedZnd opened the pull request.

ImedZnd committed:
refactoring the date class
creation on DateRange class
refactoring the EventDTO as Extension function
refactoring the find by date to be as http post insted of get in path variable
add check on the queue event handler
refactor the find by date in router as post
refactor the unittest
formatting the Initializer
formatting the RabbitMQConfiguration
match the name of the class data base repository test

ImedZnd committed:
delete unused methods

ImedZnd committed:
delete unused imports

omarbenromdhane committed:
linting

omarbenromdhane reviewed (changes requested):
fix EventRestHandler

ImedZnd committed:
adding the messages to the application.yml
ordering the params of the Date
adding the messageSource to the EventConfiguration.kt
lower the thread sleep time in the EventQueueListenerTest.kt
make the resthandler handel the i18N insted for the bad requests
refactor the router test to handle the messages in the header
addin the messages.properties in main and test packages
